Sources/core/objects/Revolver.cpp

```diff
@@ -0,0 +1,67 @@
+#include "Revolver.h"
+
+Revolver::Revolver(size_t id) : Item(id)
+{
+    SetSprite("icons/guns.dmi");
+    SetState("revolver");
+
+    name = "Revolver",
+    
+    ammunition_ = 6;
+}
+
+bool Revolver::Handle_Ammo()
+{
+    if (ammunition_ <= 0)
+    {
+        ammunition_ = 0;
+        return false;
+    }
+    --ammunition_;
+    return true;
+}
+
+void Revolver::Shoot(sf::Vector2i target)
+{	  
+    if(Handle_Ammo())
+    {
+        id_ptr_on<Projectile> p = Create<Projectile>(Bullet::T_ITEM_S(),GetOwner());
```

Comment:
I'm sure the way I made the makemovementpattern() the coordinate which I get from the targettileloc() are not always integers they can be float too.When you click you get the coordinate where did you click and check if there is something and if it's in your cantouch() range then you pick it up?I mean do you detect collision between the clicking point and the objects?